packages/runtime/container-runtime/src/test/garbageCollection.spec.ts

```diff
@@ -104,266 +68,583 @@ describe("Garbage Collection Tests", () => {
         );
     };
 
-    beforeEach(async () => {
-        mockLogger = new MockLogger();
+    describe("Inactive events", () => {
+        const inactiveObjectRevivedEvent = "GarbageCollector:inactiveObjectRevived";
+        const inactiveObjectChangedEvent = "GarbageCollector:inactiveObjectChanged";
+
+        // Waits for > deleteTimeoutMs. To be called to make sure that any unreferenced nodes have been deleted.
+        async function waitForDeleteTimeout(): Promise<void> {
+            await new Promise<void>((resolve) => {
+                setTimeout(resolve, deleteTimeoutMs + 100);
+            });
+        }
+
+        // Validates that no inactive event has been fired.
+        function validateNoInactiveEvents() {
+            assert(
+                !mockLogger.matchAnyEvent([
+                    { eventName: inactiveObjectRevivedEvent },
+                    { eventName: inactiveObjectChangedEvent },
+                ]),
+                "inactive object events should not have been logged",
+            );
+        }
+
+        // Simulates node changed activity for all the nodes in the graph.
+        function changeAllNodes(garbageCollector: IGarbageCollector) {
+            nodes.forEach((nodeId) => {
+                garbageCollector.nodeChanged(nodeId);
+            });
+        }
+
+        // Returns a dummy snapshot tree to be built upon.
+        const getDummySnapshotTree = (): ISnapshotTree => {
+            return {
+                id: "dummy",
```

Comment:
It would be great not to have IDs as I think we should delete it from ISnapshotTree (as there is no usage for it).
So, I marked it as optional some time ago but did not get to deleting it. Would be great for our code not to add IDs.